### PR TITLE
CI: Add bumpversion.yml workflow as a webhook target

### LINKED ISSUE #119: Synchronizing the number part of kolibri-explore-plugin's release into APPX's version number
This issue comes from the discussion: https://github.com/endlessm/kolibri-explore-plugin/pull/564#discussion_r1217962805

Here is the background information:
* Microsoft defines MSIX/APPX's version number as `Major.Minor.Build.Revision`.
https://github.com/microsoft/WindowsAppSDK/blob/main/specs/Deployment/MSIXPackageVersioning.md#122-version-number--msix
* Endless Key app (Android) has the version like Ladybird 6.2-329
  * "Ladybird" comes from the [VERSION file of kolibri-explore-plugin](https://github.com/endlessm/kolibri-explore-plugin/blob/master/kolibri_explore_plugin/VERSION)
  * 6.2 is the [tag of kolibri-explore-plugin](https://github.com/endlessm/kolibri-explore-plugin/tags) minus the patch (we don't use patch number)
  * 329 is the Android specific version

For Endless Key app (Windows), we can update versioning:
* sync the Major and Minor as kolibri-explore-plugin's tag minus the patch
* Build: increase number simply for Windows app build
* Revision: is `0`

### PR DESCRIPTION
**Pull request by starnight**
Add bumpversion.yml workflow as a webhook target/entry to bump release version. It accepts a new kolibri-explore-plugin release's trigger now. When it gets a new kolibri-explore-plugin release, it will append a dummy commit, then invoke the tag.yml workflow to create a new tag.

https://phabricator.endlessm.com/T34806

### CODE REVIEW

**Comment by wjt** [comment]
Again I will leave reviewing this to @dbnicholson – hope  that's OK.

**Review by manuq — APPROVED** [review]
Looks good to me, Nothing to comment.

File: .github/workflows/bumpversion.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Bump_Version
+
+on:
+  repository_dispatch:
+    types: [kolibri-explore-plugin]
```
Comment (dbnicholson) [inline .github/workflows/bumpversion.yml]: I think the type should probably called `kolibri-explore-plugin-release` just to be a bit more specific than "something happened on kolibri-explore-plugin".

File: .github/workflows/bumpversion.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # When there are multiple matched tags at the same commit, "git describe"
+      # only gets the first matched tag, not the latest one. That leads generate
+      # conflicted tag next time, due to the same tag name. So, append a dummy
+      # commit here.
+      - name: Append a new commit for new kolibri-explore-plugin
+        if: ${{ github.event.action == 'kolibri-explore-plugin' }}
```
Comment (dbnicholson) [inline .github/workflows/bumpversion.yml]: Is `github.event.action` really going to be `kolibri-explore-plugin`? It seems like you want `github.event.event_type` per https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#repository_dispatch.

**Review by dbnicholson — CHANGES_REQUESTED** [review]
This seems to make sense. I had a couple comments, but I want to review the kolibri-explore-plugin side, too.

File: .github/workflows/bumpversion.yml (hunk 2 above)
Comment (starnight) [reply .github/workflows/bumpversion.yml]: I used `github.event.event_type` at first try, but it showed nothing.  Then, did some study and found it is kind of `github.event.action`.
Here is my repo's example:
* Code: https://github.com/starnight/build-image/blob/v0.0.15/.github/workflows/bumpversion.yml#L27
* Debug run: https://github.com/starnight/build-image/actions/runs/5208911111/jobs/9398076874#step:4:18

File: .github/workflows/bumpversion.yml (hunk 1 above)
Comment (starnight) [reply .github/workflows/bumpversion.yml]: ok!

Comment (starnight) [reply .github/workflows/bumpversion.yml]: Updated by the new commit.

**Comment by starnight** [comment]
Lesson learned from https://github.com/endlessm/kolibri-explore-plugin/issues/589#issuecomment-1588459916
So, add a new commit ("CI: Fix actions/checkout's ref for bumping version").

File: .github/workflows/bumpversion.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          git commit --allow-empty -m "Bump kolibri-explore-plugin version to $VERSION"
+        env:
+          VERSION: ${{ github.event.client_payload.VERSION }}
+
+
+  create_tag:
+    needs: bump-version
+    uses: ./.github/workflows/tag.yml
+    with:
+      kep_verison: ${{ github.event.client_payload.VERSION }}
```
Comment (dbnicholson) [inline .github/workflows/bumpversion.yml]:
```suggestion
      kep_version: ${{ github.event.client_payload.VERSION }}
```

File: .github/workflows/tag.yml [hunk 4]
```diff
@@ -3,47 +3,84 @@ name: Tag Weekly
 on:
   schedule:
     - cron: '32 18 * * SAT'
+  workflow_call:
+    inputs:
+      kep_verison:
```
Comment (dbnicholson) [inline .github/workflows/tag.yml]:
```suggestion
      kep_version:
```

File: .github/workflows/tag.yml [hunk 5]
```diff
@@ -3,47 +3,84 @@ name: Tag Weekly
...
           # Do not create a new tag if latest tag is at HEAD
-          if ($revision -ne 0) {
+          if ($kle_updated -or $revision -ne 0) {
             $tag_str = "v" + $major + "." + $minor + "." + ($build + 1)
           } else {
             $tag_str = ""
           }
           echo "tag_string=$tag_str" >> $Env:GITHUB_OUTPUT
+        env:
+          kep_version: ${{ inputs.kep_verison }}
```
Comment (dbnicholson) [inline .github/workflows/tag.yml]:
```suggestion
          kep_version: ${{ inputs.kep_version }}
```

File: .github/workflows/tag.yml [hunk 6]
```diff
@@ -3,47 +3,84 @@ name: Tag Weekly
...
+              $major = $kep_major
+              $minor = $kep_minor
+              $build = 0
+              $kep_updated = 1
+            }
+          }
+
           # Do not create a new tag if latest tag is at HEAD
-          if ($revision -ne 0) {
+          if ($kle_updated -or $revision -ne 0) {
```
Comment (dbnicholson) [inline .github/workflows/tag.yml]:
```suggestion
          if ($kep_updated -or $revision -ne 0) {
```

File: .github/workflows/tag.yml [hunk 7]
```diff
@@ -3,47 +3,84 @@ name: Tag Weekly
...
+          kep_version: ${{ inputs.kep_verison }}
 
       - name: Create a new tag
         if: steps.have_new_tag_string.outputs.tag_string
-        # TODO: https://github.com/negz/create-tag/issues/14
-        uses: negz/create-tag@v1
-        with:
+        run: |
+          git tag -a $version -m $message
+          git push origin $version
```
Comment (dbnicholson) [inline .github/workflows/tag.yml:86]: I think this workflow needs the same `git config user.name` setup if it's starting from `workflow_dispatch`.

**Review by dbnicholson — CHANGES_REQUESTED** [review]
Sorry, I forgot about this PR. I just found a few typos but otherwise it looks reasonable.

File: .github/workflows/bumpversion.yml (hunk 3 above)
Comment (starnight) [reply .github/workflows/bumpversion.yml]: oh!  Good catch!  thanks!

File: .github/workflows/tag.yml (hunk 4 above)
Comment (starnight) [reply .github/workflows/tag.yml]: oh!  Good catch!  thanks!

File: .github/workflows/tag.yml (hunk 5 above)
Comment (starnight) [reply .github/workflows/tag.yml]: oh!  Good catch!  thanks!

File: .github/workflows/tag.yml (hunk 6 above)
Comment (starnight) [reply .github/workflows/tag.yml]: oh!  Good catch!  thanks!

File: .github/workflows/tag.yml (hunk 7 above)
Comment (starnight) [reply .github/workflows/tag.yml:86]: ok!

**Review by dbnicholson — APPROVED** [review]
We'll see how it goes!

**Comment by starnight** [comment]
[Bump_Version](https://github.com/endlessm/endless-key-app/actions/workflows/bumpversion.yml) is triggered by [Build for release / trigger_endless-key-app](https://github.com/endlessm/kolibri-explore-plugin/actions/runs/5383904247/jobs/9771639359#logs) successfully
https://github.com/endlessm/endless-key-app/actions/runs/5384111696

**Pull request merged**